packages/frontend-2/plugins/dayjs.ts

```diff
@@ -7,4 +7,44 @@ export default defineNuxtPlugin(() => {
   dayjs.extend(relativeTime)
   dayjs.extend(localizedFormat)
   dayjs.extend(duration)
+
+  const customRelativeTime = (date: string): string => {
+    const pastDate = dayjs(date)
+    const now = dayjs()
+    const diffInMinutes = now.diff(date, 'minute')
+    const diffInHours = now.diff(date, 'hour')
+    const diffInDays = now.diff(date, 'day')
+
+    if (diffInDays > 14) {
+      return pastDate.year() === now.year()
+        ? pastDate.format('MMM D')
+        : pastDate.format('MMM D, YYYY')
+    } else if (diffInDays >= 1) {
+      return diffInDays === 1 ? '1 day ago' : `${diffInDays} days ago`
+    } else if (diffInHours <= 23 && diffInHours >= 1) {
+      return diffInHours === 1 ? '1 hour ago' : `${diffInHours} hours ago`
+    } else if (diffInMinutes <= 59 && diffInMinutes >= 1) {
+      return diffInMinutes === 1 ? '1 minute ago' : `${diffInMinutes} minutes ago`
+    }
+
+    return 'Just now'
+  }
+
+  const isClockUnit = (date: string) => {
+    const unit = customRelativeTime(date)
+    return unit.includes('second') || unit.includes('minute') || unit.includes('hour')
+  }
+
+  const getFullDate = (input: string) => dayjs(input).format('MMM D, YYYY, H:mm')
+  const getTrunicatedRelativeDate = (input: string) => customRelativeTime(input)
+  const getTrunicatedRelativeDateWithPrefix = (input: string) =>
+    isClockUnit(input) ? customRelativeTime(input) : `on ${customRelativeTime(input)}`
+
+  return {
+    provide: {
```

Comment:
Not a fan of this being a nuxt plugin - this means you can only ever use these in Vue scope or you have to get these from the NuxtApp instance.

Since the dayjs import is a singleton (you can import dayjs(), you don't need to work with this specific instance in this plugin), you could just have all of these functions be importable helpers. Better yet - if you define these somewhere under the `utils/` folder, they're gonna be auto-imported everywhere so that you don't even need to type out any imports.

Then you also don't need to have the dollar sign prefix for these which is kind of annoying too

Reply:
Sure I can move them to the utils folder 👍